yt_dlp/utils/_utils.py

```diff
@@ -5638,6 +5116,39 @@ def truncate_string(s, left, right=0):
     return f'{s[:left-3]}...{s[-right:] if right else ""}'
 
 
+class CaseInsensitiveDict(collections.UserDict):
+    """
+    Store and access keys case-insensitively.
+    The constructor can take multiple dicts, in which keys in the latter are prioritised.
+    """
+    def __init__(self, *args, **kwargs):
+        super().__init__()
+        for dct in args:
+            if dct is None:
+                continue
+            self.update(dct)
+        self.update(kwargs)
+
+    def __setitem__(self, key, value):
+        super().__setitem__(key.title(), str(value))
+
+    def __getitem__(self, key):
+        return super().__getitem__(key.title())
+
+    def __delitem__(self, key):
+        super().__delitem__(key.title())
+
+    def __contains__(self, key):
+        return super().__contains__(key.title() if isinstance(key, str) else key)
+
+
+def infojson_decoder_hook(data):
+    # HTTP Headers. Assuming user-agent is in infojson headers.
+    if isinstance(data, dict) and 'user-agent' in [k.lower() for k in data.keys()]:
```

Comment:
go by key rather than contents maybe

Reply:
Hmm yeah maybe it is best to leave this out of this PR for now since it's not required